From b8abee38c8362b6708e638dc01d8788f69afdf2b Mon Sep 17 00:00:00 2001
From: Vibhore Vardhan <vibhore@ti.com>
Date: Wed, 5 Feb 2025 14:27:57 -0600
Subject: [PATCH 06/68] TI: HACK: arm64: dts: ti: Configure am62p
 pinctrl-single for wake-up interrupts

Configure am62p pinctrl-single for wake-up interrupts.

Based on the patch for am625. No changes to the length of the register
set and the interrupt ID were needed for am62p as they match those on
am625.

Marked as HACK because physically none of the wakeup source events are
connected to GIC. Eventually we may want to set up a chained interrupt
handler in the ti-msgmgr for pinctrl-single to use.

[k-willis@ti.com] appended compatible with single-pinctrl to fix DT
warning. Squashed relevant  part of commit eb3b9e22be86 ("arm64: dts: ti:
k3-am62*: Append pinctrl-single compatible").

Signed-off-by: Vibhore Vardhan <vibhore@ti.com>
Signed-off-by: Dhruva Gole <d-gole@ti.com>
Signed-off-by: Devarsh Thakkar <devarsht@ti.com>
Reported-by: kernel test robot <lkp@intel.com>
Reviewed-by: Dhruva Gole <d-gole@ti.com>
Signed-off-by: Kendall Willis <k-willis@ti.com>
---
 .../boot/dts/ti/k3-am62p-j722s-common-main.dtsi   | 15 ++++++++++++++-
 1 file changed, 14 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/ti/k3-am62p-j722s-common-main.dtsi b/arch/arm64/boot/dts/ti/k3-am62p-j722s-common-main.dtsi
index 9b813d2485e5..808b02f87089 100644
--- a/arch/arm64/boot/dts/ti/k3-am62p-j722s-common-main.dtsi
+++ b/arch/arm64/boot/dts/ti/k3-am62p-j722s-common-main.dtsi
@@ -254,11 +254,24 @@ secure_proxy_sa3: mailbox@43600000 {
 	};
 
 	main_pmx0: pinctrl@f4000 {
-		compatible = "pinctrl-single";
+		compatible = "ti,am654-padconf", "pinctrl-single";
 		reg = <0x00 0xf4000 0x00 0x2ac>;
 		#pinctrl-cells = <1>;
 		pinctrl-single,register-width = <32>;
 		pinctrl-single,function-mask = <0xffffffff>;
+		/*
+		 * pinctrl IP DOES NOT give any functional IRQs when a
+		 * system is in active state. This IRQ is only a dummy IRQ
+		 * that is being used to trick Linux into thinking that
+		 * GIC can potentially recieve an interrupt from this IP.
+		 * This helps us setup the IO daisychain wakeups for deep
+		 * sleep via chained wake IRQs.
+		 * Please feel free to assign a different number here as
+		 * long as it is unused if 98 conflicts with another use case.
+		 */
+		interrupts = <GIC_SPI 98 IRQ_TYPE_LEVEL_HIGH>;
+		interrupt-controller;
+		#interrupt-cells = <1>;
 		bootph-all;
 	};
 
-- 
2.47.2

